### PR TITLE
feat(plugin-meetings): get screen and window size in mqe

### PR DESCRIPTION
**Pull request by edvujic**
# COMPLETES [WEBEX-379503](https://jira-eng-gpk2.cisco.com/jira/browse/WEBEX-379503)

## This pull request addresses

The need for more accurate tracking and reporting of window and screen sizes in our analytics in MQRs.

## by making the following changes

- Added new properties to `newMqa.intervalMetadata` to capture the screen and app window dimensions:
  - `screenWidth` and `screenHeight` to store the dimensions of the screen in pixels.
  - `screenResolution` to store the screen resolution in blocks of 16x16 pixels.
  - `appWindowWidth` and `appWindowHeight` to store the dimensions of the app window in pixels.
  - `appWindowSize` to store the app window size in blocks of 16x16 pixels.
- These properties are used to populate `report_intervalMetadata_screenResolution` and `report_intervalMetadata_appWindow(Height/Width/Size)` for MQRs.
- Updated the test suite to include tests that verify the correct assignment of these properties based on `window.screen` and `window` properties.

# Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

# The following scenarios were tested

- Ensured that `screenWidth`, `screenHeight`, and `screenResolution` are correctly calculated and assigned based on the `window.screen` object.
- Verified that `appWindowWidth`, `appWindowHeight`, and `appWindowSize` are correctly calculated and assigned based on the window object.
- All tests were performed using automated unit tests with sinon stubs to mock window properties.

# I certified that

- [X] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [X] I discussed changes with code owners prior to submitting this pull request
- [X] I have not skipped any automated checks
- [X] All existing and new tests passed
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: packages/@webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js [hunk 1]
```diff
@@ -1862,6 +1862,41 @@ describe('plugin-meetings', () => {
...
+          sinon.stub(window.screen, 'width').get(() => 1280);
+          sinon.stub(window.screen, 'height').get(() => 720);
+          await progressTime(MQA_INTERVAL);
+          assert.strictEqual(mqeData.intervalMetadata.screenWidth, 1280);
+          assert.strictEqual(mqeData.intervalMetadata.screenHeight, 720);
+          assert.strictEqual(mqeData.intervalMetadata.screenResolution, 3600);
+        })
+
+        context('when app window size changes', () => {
+          it('should record the initial app window size from window properties', async () => {
```
Comment (sreenara) [inline packages/@webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js:1881]: These 2 tests seem to be the same.. in order to test if window size changes are being recorded, would it be better if we do this within one test itself where the `get()` function returns different values for the first and second times it's called?

**Review by sreenara — COMMENTED** [review]
1. Question about the UT.
2. In the description, "tooling change" has also been checked, but I don't see that. Am I missing something?
3. Was there manual testing done with the web client for the tests mentioned in the PR description?

**Review by brycetham — APPROVED** [review]
Looks great!

**Comment by edvujic** [comment]
> 1. Question about the UT.
> 2. In the description, "tooling change" has also been checked, but I don't see that. Am I missing something?
> 3. Was there manual testing done with the web client for the tests mentioned in the PR description?

Hi @sreenara, thought that statsAnalyzer is a tool that we use to send info to our internal troubleshooting tools, thus I thought it would be appropriate.

Yes, manual testing with the web client has been done.

**Review by sreenara — APPROVED** [review]

**Pull request merged**